server/pages/index.js

```diff
@@ -204,6 +206,10 @@ function getDefaultContext( request ) {
 		sectionCss = request.context.sectionCss;
 	}
 
+	if ( request.context && request.context.lang ) {
```

Comment:
Since the way that middleware passes `context` around can be a bit opaque, maybe consider adding a comment that `request.context.lang` is set by the `handleLocaleSubdomains()` middleware.